src/test/test-notify-recv.c

```diff
@@ -80,7 +91,17 @@ TEST(notify_socket_prepare) {
         if (r == 0) {
                 ASSERT_OK_ERRNO(setenv("NOTIFY_SOCKET", path, /* overwrite = */ true));
                 ASSERT_OK_POSITIVE(sd_notify(/* unset_environment = */ false, "FIRST_MESSAGE=1"));
-                ASSERT_OK_POSITIVE(sd_notify(/* unset_environment = */ false, "FIRST_MESSAGE=2\nADDITIONAL_DATA=hoge"));
+
+                _cleanup_close_ int fd1 = open("/tmp", O_RDONLY|O_CLOEXEC|O_DIRECTORY);
+                _cleanup_close_ int fd2 = open("/dev/null", O_RDONLY|O_CLOEXEC);
+                ASSERT_TRUE(fd1 >= 0);
+                ASSERT_TRUE(fd2 >= 0);
+
+                int fds[] = {fd1, fd2};
+                ASSERT_OK_POSITIVE(
+                        sd_pid_notify_with_fds(
+                                0, /* unset_environment = */ false,
+                                "FIRST_MESSAGE=2\nADDITIONAL_DATA=hoge", fds, 2));
```

Comment:
It is not necessary to introduce `fds`, you can directly pass them like the following:
```c
sd_pid_notify_with_fds(
        ...,
        (int[]) { fd1, fd2 }, 2);
```